File: Backend/src/main/java/org/example/backend/dto/SubjectSubjectRequirementDto.java
package org.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Value;

import java.io.Serializable;

/**
 * DTO for {@link org.example.backend.domain.SubjectSubjectRequirement}
 */
@Data
@NoArgsConstructor
public class SubjectSubjectRequirementDto implements Serializable {
    SubjectSubjectRequirementIdDto id;
    SubjectDto subject;
    SubjectRequirementDto subjectRequirement;
}

File: Frontend/src/app/dto/subject-subject-requirement-dto.ts
import {SubjectSubjectRequirementIdDto} from "./subject-subject-requirement-id-dto";
import {SubjectDto} from "./subject-dto";
import {SubjectRequirementDto} from "./subject-requirement-dto";

export interface SubjectSubjectRequirementDto {
  id : SubjectSubjectRequirementIdDto
  subject : SubjectDto
  subjectRequirement : SubjectRequirementDto
}
